src/core.js

```diff
@@ -23,32 +22,65 @@ async function loadPage (page, url, timeout, pageLoadSkipTimeout) {
   // set a higher number than the timeout option, in order to make
   // puppeteer’s timeout _never_ happen
   let waitingForPageLoad = true
-  const loadPagePromise = page.goto(url, { timeout: timeout + 1000 })
+  let pageLoadSkipTimeoutPromise = null
+  const loadPagePromise = new Promise((resolve, reject) => {
+    page
+      .goto(url, {
+        timeout: timeout + 1000
+      })
+      .then(response => {
+        // Reject exits the Promise.all call immediately and ensures that the pageload is always higher prio than timout
+        return resolve('loadPageResponse')
+      })
+      .catch(err => {
+        // Reject when error because we don't want an errored page
+        console.error(err)
+        return reject(err)
+      })
+  })
+
   if (pageLoadSkipTimeout) {
-    await Promise.race([
-      loadPagePromise,
-      new Promise(resolve => {
-        // instead we manually _abort_ page load after X time,
-        // in order to deal with spammy pages that keep sending non-critical requests
-        // (tracking etc), which would otherwise never load.
-        // With JS disabled it just shouldn't take that many seconds to load what's needed
-        // for critical viewport.
-        setTimeout(() => {
-          if (waitingForPageLoad) {
-            debuglog(
-              'page load waiting ABORTED after ' +
-                pageLoadSkipTimeout / 1000 +
-                's. '
-            )
-            resolve()
+    pageLoadSkipTimeoutPromise = new Promise((resolve, reject) => {
+      // enables us to control requests(stop, continue, modify)
+      page.on('response', async response => {
```

Comment:
I don't understand this logic, and what it has to do with the pageLoadSkipTimeout... that timeout is meant to run in parallel with the page load, to exit if the page load takes to long. Moving it into a response handler defeats that purpose.

Reply:
I cant tell you why this is needed to be an wrapped promise. But I got different errors regarding a closed target. 
You can remember my investigation ins this part of the code where the response size was different when the pageLoadSkipTimeout was close the the real page load timer. This complete section solved that problem.